service/s3/s3manager/upload.go

```diff
@@ -638,6 +706,43 @@ func (u *multiuploader) send(c chunk) error {
 	return nil
 }
 
+// isChunkUploaded is called before uploading each chunk to determine
+// if this chunk can be skipped. Chunks can be skipped when resuming an upload
+// if the ETag matches for the previously uploaded chunk.
+func (u *multiuploader) isChunkUploaded(c chunk) bool {
```

Comment:
I'm not sure we can use the `ETag` to check the upload part is the same as the local content. Specifically because the `ETag` is not required to be the MD5 of the part. By convention this the `ETag` probably is the MD5, but there is not guarantee that this is so. We'd probably not want to rely on unofficial behavior of the ETag.

Reply:
I can't find good AWS documentation defining the `ETag` values for multipart uploads. Using the MD5 seems pretty standard though. Is there a better option than using the MD5?